regolith/builders/coabuilder.py

```diff
@@ -3,40 +3,211 @@
+def query_people_and_instituions(rc, names):
+    """Get the people and institutions names."""
+    people, institutions = [], []
+    for person_name in names:
+        person_found = fuzzy_retrieval(all_docs_from_collection(
+            rc.client, "people"),
+            ["name", "aka", "_id"],
+            person_name)
+        if not person_found:
+            person_found = fuzzy_retrieval(all_docs_from_collection(
+                rc.client, "contacts"),
+                ["name", "aka", "_id"], person_name)
+            if not person_found:
+                print(
+                    "WARNING: {} not found in contacts. Check aka".format(
+                        person_name))
+            else:
+                people.append(person_found['name'])
+                inst = fuzzy_retrieval(all_docs_from_collection(
+                    rc.client, "institutions"),
+                    ["name", "aka", "_id"],
+                    person_found["institution"])
+                if inst:
+                    institutions.append(inst["name"])
+                else:
+                    institutions.append(person_found.get("institution", "missing"))
+                    print("WARNING: {} missing from institutions".format(
+                        person_found["institution"]))
+        else:
+            people.append(person_found['name'])
+            pinst = person_found.get("employment",
+                                     [{"organization": "missing"}])[
+                0]["organization"]
+            inst = fuzzy_retrieval(all_docs_from_collection(
+                rc.client, "institutions"), ["name", "aka", "_id"],
+                pinst)
+            if inst:
+                institutions.append(inst["name"])
+            else:
+                institutions.append(pinst)
+                print(
+                    "WARNING: {} missing from institutions".format(
+                        pinst))
+    return people, institutions
+
+
+def query_last_first_instutition_names(rc, ppl_names, excluded_inst_name=None):
+    """Get the last name, first name and institution name."""
+    ppl = []
+    for ppl_tup in ppl_names:
+        inst = fuzzy_retrieval(
+            all_docs_from_collection(rc.client, "institutions"),
+            ['aka', 'name', '_id'], ppl_tup[1],
+            case_sensitive=False)
+        if inst:
+            inst_name = inst.get("name", "")
+        else:
+            inst_name = ppl_tup[1]
+        # remove all people who are in the institution of the person
+        if inst_name != excluded_inst_name:
+            name = HumanName(ppl_tup[0])
+            ppl.append((name.last, name.first, ppl_tup[1]))
+    ppl = list(set(ppl))
+    ppl.sort(key=lambda x: x[0])
+    return ppl
+
+
+def format_people_name(ppl_names):
+    """Format people name to be first name, last name."""
+    ppl = []
+    # reformatting the name in last name, first name
+    for idx in range(len(ppl_names)):
+        names = ppl_names[idx][0].split()
```

Comment:
please change to use HumanName methods.  I think I made this change in a PR somewhere. Yes, I checked. This makes me a bit nervous that you are making these changes on not hte latest version... This will be very difficult to rebase....

Reply:
I found that. There are no conflicts so no need for rebasing. I will use the HumanName methods too in the modified code to solve the problem.

Reply:
Hi Simon. I checked. I am modifying on your latest version. I move this part to a function and call the function inside the query method in the class.